test-suite/output/Implicit.out

```diff
@@ -12,3 +12,8 @@ map id' (1 :: nil)
      : list nat
 map (id'' (A:=nat)) (1 :: nil)
      : list nat
+fix f (x : nat) : option nat := match x with
```

Comment:
BTW why don't we put a newline before the match here?

Reply:
The `match` block is always printed in a vertical box (`Pp.v`). Apparently, here, since the vertical `match` block does not exceed the total width of the line, the OCaml format library decides not to break the line.

We could relax the `v` box used for printing `match` into an `hv` box, but the risk then, is that it produces:
```
fix f (x : nat) : option nat :=
  match x with | 0 => None nat | S _ => some_nat x end
```
which I don't know if it is better.

Or, maybe, do we want that the body of the fix is always put at the next line if it does not fit on the line. Then, something could probably be tried:
- replace the `hov 0` in the printing of `fix`/`cofix` with a `hv 0`
- protect the `binders : typ :=` part in `Ppconstr.pr_recursive_decl` with an extra `hov 0` so that its inner breaking points are not considered `v`

I believe a PR to improve that could be proposed!

Reply:
@skyskimmer: I realize that  #11563 does not improve the formatting of `fix `which it was intended to improve in this PR. See explanation in my previous comment.

I believe it would work if `match` were a `hv` box, but I don't know how to tell that a vertical bar has to be printed iff the box is vertical (and `match x with | 0 => None | S _ => x end` would not be nice).

Well, I'm renouncing to improve it for this time.

Reply:
I'm not sure that putting the match on a single line when there's more than 1 branch is good tbh, maybe we should just put each branch on its own line and print leading `|` in that case.

Reply:
We could also say, in the same way as `match` with several cases is always vertical, that `fix` always puts its body on a different line?